docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,181 @@
+is helpful, and sometimes critical, to be notified of this data 
+migration.
+
+Quilt's *Event-Driven Packaging* service (EDP) monitors S3 object events
+for a given Amazon S3 bucket prefix and groups those S3 object
+events into logical batches via custom-defined time, space and
+prefix heuristics. A `files_ready` event is generated on the receiving
+[EventBridge](https://aws.amazon.com/eventbridge/) in order to
+notify users and/or other AWS services that this logical batch of
+S3 objects are ready to be turned into Quilt packages. *This makes
+it easy to ingest data into Quilt data packages from AWS data
+migration services* such as [Storage
+Gateway](https://aws.amazon.com/storagegateway/) and [Data
+Sync](https://aws.amazon.com/datasync/).
+
+> Any AWS service or action that generates S3 object events may trigger the EDP service.
+
+## Requirements
+
+1. A pre-existing VPC that includes a [NAT
+Gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html)
+or the following [VPC
+endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html#concepts-vpc-endpoints):
+  - Amazon S3
+  ([gateway](https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html)
+  or
+  [interface](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html))
+  - EventBridge (interface)
+2. [Enable EventBridge S3
+Events](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications-eventbridge.html)
+for all S3 buckets to be monitored by EDP
+
+![](../imgs/edp-event-bridge-s3-properties.png)
+
+## Deployment
+
+To use EDP in your Quilt instance, you will need an additional
+CloudFormation template to configure when, and how, events get
+generated.
+
+### Parameters
+
+The following parameters need to be defined in the CloudFormation
+template during deployment:
+
+- `VPC`: Same as the existing Quilt instance.
+- `SecurityGroup`: Same as the existing Quilt instance.
+- `Subnets`: Same two subnets as the existing Quilt instance.
+- `BucketName`: Name of the Amazon S3 bucket to monitor. Must be
+read accessible to EDP.
+- `BucketIgnorePrefixes`: List of bucket path segments to ignore
+- `BucketPrefixDepth`: The number of `/`-separated path segments
+at the beginning of an S3 object key. Default value is `2`.
+- `BucketThresholdDuration`: Trigger a notification when this number
+of seconds has elapsed since the last object event in the S3 bucket 
+occurred. Default value is `300` seconds.
+- `BucketThresholdEventCount`: Trigger a notification when this
+number of files have been created (since the prior trigger)
+- `DBUser`: Username for EDP RDS instance
+- `DBPassword`: Password for EDP RDS instance
+- `EventBusName`: Name of custom EventBridge event bus that recieves events
+
+![](../imgs/edp-cloudformation-parameters.png)
+
+## How EDP works
+
+1. EDP monitors S3 object events for _s3://source-bucket_
+2. After a fixed number of events (`BucketThresholdEventCount`) or
```

Comment:
Does this mean the event ALWAYS fires if the threshold is exceeded, even if from multiple disjoint files with no common prefix?

Reply:
I believe so, yes, but would like confirmation from @sir-sigurd

Reply:
No. The threshold is per common prefix.

Reply:
```suggestion
a maximum duration within a common prefix (`BucketThresholdDuration`) EDP creates a
```
suggest adding "within a common prefix" to reduce ambiguity